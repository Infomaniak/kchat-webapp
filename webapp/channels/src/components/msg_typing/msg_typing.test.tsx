// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import MsgTyping from 'components/msg_typing/msg_typing';

describe('components/MsgTyping', () => {
    const baseProps = {
        typingUsers: [],
        recordingUsers: [],
        channelId: 'test',
        rootId: '',
        userStartedTyping: jest.fn(),
        userStoppedTyping: jest.fn(),
        userStartedRecording: jest.fn(),
        userStoppedRecording: jest.fn(),
    };

    test('should match snapshot, on nobody typing', () => {
        const wrapper = shallow(<MsgTyping {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on one user typing', () => {
        const typingUsers = ['test.user'];
        const props = {...baseProps, typingUsers};

        const wrapper = shallow(<MsgTyping {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on multiple users typing', () => {
        const typingUsers = ['test.user', 'other.test.user', 'another.user'];
        const props = {...baseProps, typingUsers};

        const wrapper = shallow(<MsgTyping {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should should match snapshot, on nobody recording', () => {
        const wrapper = shallow(<MsgTyping {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on one user recording', () => {
        const recordingUsers = ['test.user'];
        const props = {...baseProps, recordingUsers};

        const wrapper = shallow(<MsgTyping {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on multiple users recording', () => {
        const recordingUsers = ['test.user', 'other.test.user', 'another.user'];
        const props = {...baseProps, recordingUsers};

        const wrapper = shallow(<MsgTyping {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
