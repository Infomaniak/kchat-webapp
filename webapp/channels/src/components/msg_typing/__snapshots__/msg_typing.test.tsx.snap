// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MsgTyping should match snapshot, on multiple users recording 1`] = `
<span
  className="msg-typing"
>
  <MemoizedFormattedMessage
    defaultMessage="{users} and {last} are recording..."
    id="msg_recording.areRecording"
    values={
      Object {
        "last": "another.user",
        "users": "test.user, other.test.user",
      }
    }
  />
</span>
`;

exports[`components/MsgTyping should match snapshot, on multiple users typing 1`] = `
<span
  className="msg-typing"
>
  <MemoizedFormattedMessage
    defaultMessage="{users} and {last} are typing..."
    id="msg_typing.areTyping"
    values={
      Object {
        "last": "another.user",
        "users": "test.user, other.test.user",
      }
    }
  />
</span>
`;

exports[`components/MsgTyping should match snapshot, on nobody typing 1`] = `
<span
  className="msg-typing"
/>
`;

exports[`components/MsgTyping should match snapshot, on one user recording 1`] = `
<span
  className="msg-typing"
>
  <MemoizedFormattedMessage
    defaultMessage="{user} is recording..."
    id="msg_recording.isRecording"
    values={
      Object {
        "user": "test.user",
      }
    }
  />
</span>
`;

exports[`components/MsgTyping should match snapshot, on one user typing 1`] = `
<span
  className="msg-typing"
>
  <MemoizedFormattedMessage
    defaultMessage="{user} is typing..."
    id="msg_typing.isTyping"
    values={
      Object {
        "user": "test.user",
      }
    }
  />
</span>
`;

exports[`components/MsgTyping should should match snapshot, on nobody recording 1`] = `
<span
  className="msg-typing"
/>
`;
